typed_sql/lib/src/dialect/sqlite_dialect.dart

```diff
@@ -0,0 +1,624 @@
+  (String, List<Object?>) insertInto(InsertStatement statement) {
+    final (params, ctx) = QueryContext.create();
+
+    var returnProjection = '';
+    final returning = statement.returning;
+    if (returning != null) {
+      final c = ctx.scope(returning, returning.columns);
+      returnProjection = returning.projection.map((e) => expr(e, c)).join(', ');
+    }
+
+    // TODO: Is it possible to insert a ModelExpression
+    //       It probably is, if copying a row from one table to another!
+    return (
+      [
+        'INSERT INTO ${escape(statement.table)}',
+        '(${statement.columns.map(escape).join(', ')})',
+        'VALUES (${statement.values.map((e) => expr(e, ctx)).join(', ')})',
+        if (returnProjection.isNotEmpty) 'RETURNING $returnProjection',
+      ].join(' '),
+      params,
+    );
+  }
+
+  @override
+  (String, List<Object?>) update(UpdateStatement statement) {
+    final (params, ctx) = QueryContext.create();
+    final (alias, c) = ctx.alias(statement, statement.table.columns);
+    final (sql, _) = clause(statement.where, ctx);
+
+    var returnProjection = '';
+    final returning = statement.returning;
+    if (returning != null) {
+      final c = ctx.scope(returning, returning.columns);
+      returnProjection = returning.projection.map((e) => expr(e, c)).join(', ');
+    }
+
+    return (
+      [
+        'UPDATE ${escape(statement.table.name)} AS $alias',
+        'SET',
+        statement.columns
+            .mapIndexed(
+              (i, column) => '$column = (${expr(statement.values[i], c)})',
+            )
+            .join(', '),
+        'WHERE EXISTS (SELECT TRUE FROM ($sql) AS _subq WHERE',
+        statement.table.primaryKey
+            .map((f) => '$alias.${escape(f)} = _subq.${escape(f)}')
+            .join(', '),
+        ')',
+        if (returnProjection.isNotEmpty) 'RETURNING $returnProjection',
+      ].join(' '),
+      params,
+    );
+  }
+
+  @override
+  (String, List<Object?>) delete(DeleteStatement statement) {
+    final (params, ctx) = QueryContext.create();
+    final (sql, _) = clause(statement.where, ctx);
+
+    var returnProjection = '';
+    final returning = statement.returning;
+    if (returning != null) {
+      final c = ctx.scope(returning, returning.columns);
+      returnProjection = returning.projection.map((e) => expr(e, c)).join(', ');
+    }
+
+    return (
+      [
+        'DELETE FROM ${escape(statement.table.name)} as _topq',
+        'WHERE EXISTS (SELECT TRUE FROM ($sql) AS _subq WHERE',
+        statement.table.primaryKey
+            .map((f) => '_topq.${escape(f)} = _subq.${escape(f)}')
+            .join(', '),
+        ')',
+        if (returnProjection.isNotEmpty) 'RETURNING $returnProjection',
+      ].join(' '),
+      params,
+    );
+  }
+
+  /// Escape [name] for use as identifier in SQL.
+  ///
+  /// Example: 'my string' -> '"my string"'
+  ///
+  /// Throws if [name] cannot be safely escaped.
+  String escape(String name) {
```

Comment:
Perhaps make these private (unless they are supposed to be used outside the specific dialiect file.

Reply:
Oh - sorry - thought they were top-level. Might need new glasses :)